src/corehost/common/pal.unix.cpp

```diff
@@ -183,6 +183,20 @@ bool pal::get_own_executable_path(pal::string_t* recv)
     }
     return false;
 }
+#elif defined(__FreeBSD__)
+bool pal::get_own_executable_path(pal::string_t* recv)
+{
+    int mib[4];
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PATHNAME;
+    mib[3] = -1;
+    char buf[1024];
```

Comment:
We should not ignore the error code from `sysctl`. Handle `ENOMEM` as well?

Some nits:
a `static const int mib[] = { };` 
Use `_countof` or `__arraycount` macros
`PATH_MAX` instead of 1024